Source/WebCore/dom/EventDispatcher.cpp

```diff
@@ -186,7 +186,8 @@ void EventDispatcher::dispatchEvent(Node& node, Event& event)
     RefPtr inputForLegacyPreActivationBehavior = dynamicDowncast<HTMLInputElement>(node);
     if (!inputForLegacyPreActivationBehavior && event.bubbles() && event.type() == eventNames().clickEvent)
         inputForLegacyPreActivationBehavior = findInputElementInEventPath(eventPath);
-    if (inputForLegacyPreActivationBehavior)
+    if (inputForLegacyPreActivationBehavior
+        && (!event.isTrusted() || !inputForLegacyPreActivationBehavior->isDisabledFormControl()))
         inputForLegacyPreActivationBehavior->willDispatchEvent(event, clickHandlingState);
```

Comment:
I believe coding style says to use curly brackets if the condition is on several lines.